scripts/build/bin/procs.sh

```diff
@@ -45,21 +58,29 @@ elif [ "$COMMAND" == "ROVERLAPS"  ]; then
     rm -f $TMP
 
 elif [ "$COMMAND" == "GITPACK"  ]; then
-    # set the remote to the writeable url
-    git remote set-url origin  ssh://p-mu2eofflinesoftwaremu2eoffline@cdcvs.fnal.gov/cvs/projects/mu2eofflinesoftwaremu2eoffline/Offline.git
-    [ $? -ne 0 ] && RC=1
-    # make sure .git is packed
+    if [ -n "$MUSE_WORK_DIR"  ]; then
```

Comment:
I'm surprised to see muse-specific references here.  I thought Muse was a layer on top of Offline.  Will analysis repos also need to have muse-specific SConstruct?

Reply:
Muse is a layer over Offline build for all code because code is built using scons patterns (cc->.os) and a very a simple and predictable directory structure which scons is designed to handle.  genTriggerFcl and procs.sh both write output to arbitrary directories, not in any standard build pattern.  I chose to make these pieces do either standard build or Muse build so after the transition, we remove standard build pattern and it appear "simple" again. There are two other possibilities that I can think of.  Run the standard build, let it write to gen directory in Offline, them mv that directory to the out of source build area.  I don't like this since gen might be left in the Offline area and it is also a really unexpected pattern. Another approach would be to teach scons more one-off build patterns where it can contain the paths.  This is a possibility, but I'd have to think about it.  If we had to live with both forever, I'd be more motivated to do that.  What I did keeps the build pattern of the scripts writing to arbitrary areas.  Also note that Muse is doing more than the Offline build in that it can write multiple parallel builds, so while Offline build writes one place, muse writes many, so that factor must be introduced, making this script more complex and Muse-aware.   So if I re-wrote the fundamentals of how these things build I think I could make it cleaner/slicker.  I will add it to my list as a low priority, since what is here is working fine with both builds.

Reply:
I am fine with there being some cruft during this transition.  Mostly this got me thinking about muse requirements for analysis repos: what env variables will have to be defined and respected during the build?  How should MUSE_WORK_DIR etc be interpreted?  Will you provide an scons template?   Will muse support other make systems?  If so, are the hooks well defined?

Reply:
I like the way you are doing it now.  I prefer that to running the standard build and moving things around.  With your way, you can do the real developement/debugging of Muse now - with the other way some of the development is necessarily  deferred until you disable the old style build - that seems risky to me.

I am committed to removing support for non-Muse builds as soon as Muse is reasonably stable.

Reply:
Following up on Dave's comment.  I had asked Ray a question privately but I will repeat it here.  What is the plan for building/Running the Tutorial repo against an externally supplied version of Offline.  I think that this will serve as a model for doing analysis?  Dave - is that what you were getting at?

Reply:
For the functions in procs.sh, it might be best to move them to codeTools repo since it is more packaging of Offline than building.  I think I favor this, so let me know if you don't like it.  FYI I'm also thinking on providing a UPS setup of Offline release on cvmfs.

Reply:
I went into this assuming a baseline functionality is to build a local analysis repos against a local Offline, a linked Offline outside of this build area (cvmfs or another area), and a pgit partial build with a backing linked Offline.  The first is done, the others are sketched out and a high priority to finish.